<a href="https://colab.research.google.com/github/KartheekAnusuri/Applied-Tech/blob/main/Project18.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Instructions

#### Goal of the Project

This project is designed for you to practice and solve the activities that are based on the concepts covered in the following lessons:

1. While Loop, Data-Type Conversion and Conditional Statements.

2. Python Lists (`for` Loop).


---

---

### Activities


  

#### Activity 1: Last Digit Ultimate

Your job is to create a function, that takes 3 numbers: `num1, num2, num3` and returns `True` if:

 (last digit of `num1`) * (the last digit of `num2`) = the last digit of `num3`.

**Examples**
```
1. last_dig(25, 21, 125) ➞ True
# The last digit of 25 is 5, the last digit of 21 is 1,
# last digit of 125 is 5, and the last digit of 5*1 = 5,
# which is equal to the last digit of 125 i.e. 5

2. last_dig(55, 226, 5190) ➞ True
# The last digit of 55 is 5, the last digit of 226 is 6,
# last digit of 5190 is 0, and the last digit of 5*6 = 30 i.e. 0,
# which is equal to the last digit of 5190 i.e. 0

3. last_dig(12, 215, 2142) ➞ False
# The last digit of 12 is 2, the last digit of 215 is 5,
# last digit of 2142 is 2, and the last digit of 2*5 = 10 is 0,
# which is not equal to the last digit of 2142 i.e. 2
```


In [ ]:
# Write your solution here:
def last_dig(num1, num2, num3):
    # Step 1: Get the last digit of each number
    last_digit_num1 = num1 % 10
    last_digit_num2 = num2 % 10
    last_digit_num3 = num3 % 10

    # Step 2: Calculate the product of the last digits
    product_last_digits = last_digit_num1 * last_digit_num2

    # Step 3: Get the last digit of the product
    last_digit_product = product_last_digits % 10

    # Step 4: Compare the last digit of the product with the last digit of num3
    return last_digit_product == last_digit_num3

# Test cases
print(last_dig(25, 21, 125))
print(last_dig(55, 226, 5190))
print(last_dig(12, 215, 2142))

True
True
False


---

#### Activity 2: Pentagonal Number

Given a number $num$ as a positive integer, calculate how many dots exist in a pentagonal shape around the centre dot on the $Nth$ iteration.

In the image below you can see in the first iteration there is only a single dot. On the second, there are 6 dots. On the third, there are 16 dots, and on the fourth there are 31 dots.

<img src='https://student-datasets-bucket.s3.ap-south-1.amazonaws.com/images/pengatonal-number.png' width=500>

**Constraint**

$num > 0$

**Sample Input 1**
```
2
```

**Sample Output 1**
```
6
```

**Sample Input 2**
```
3
```

**Sample Output 2**
```
16
```


In [ ]:
# Write your solution here
def pentagonal_number(num):
    # Handle the base case where num is 1 (first iteration)
    if num == 1:
        return 1

    # Calculate the number of dots for the Nth iteration
    dots = 1 + sum(5 * i for i in range(1, num))

    return dots

# Test cases
print(pentagonal_number(2))
print(pentagonal_number(3))
print(pentagonal_number(9))

6
16
181


**Q**: How many dots are there in the 9th iteration?

**A**: 181

---

---